src/java/main/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -325,6 +327,8 @@ public void parseProperties(Properties zkProp)
                 quorumServicePrincipal = value;
             } else if (key.equals("quorum.cnxn.threads.size")) {
                 quorumCnxnThreadsSize = Integer.parseInt(value);
+            } else if (key.equals("metricsProvider.className")) {
```

Comment:
In the case of the server side there is no problem.
For the client, you can have multiple zk clients in the same jvm.
It would be better to have a single way of configuring metrics

Reply:
For instance Prometheus may expose a configuration parameter for the URI to bind to (it will default to '/metrics').
Other systems may need authentication parameters.

Reply:
@eolivelli are you saying we want to keep the same config name in static file because we may want to reuse the same config on client side as well?